### PR TITLE
state tree

### PR DESCRIPTION
**Pull request by turuslan**
- remove excessive "state_tree" dependencies
- remove unused "genesis_test"

### CODE REVIEW

File: core/vm/actor/builtin/states/init_actor_state_raw.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#include "adt/address_key.hpp"
+#include "storage/hamt/hamt.hpp"
+#include "vm/actor/codes.hpp"
+
+namespace fc::vm::actor {
+  struct InitActorStateRaw {
+    using Address = primitives::address::Address;
+    using Hamt = storage::hamt::Hamt;
+
+    CID _address_map;
```
Comment (Alexey-N-Chernyshov) [inline core/vm/actor/builtin/states/init_actor_state_raw.hpp:17]:
```suggestion
    CID address_map_cid;
```

File: test/core/vm/state/state_tree_test.cpp [hunk 2]
```diff
@@ -87,9 +87,15 @@ TEST_F(StateTreeTest, Walk) {
   adt::Map<vm::actor::Actor, adt::AddressKeyer> map{store_};
   auto head1{store_->setCbor(3).value()};
   auto code1{vm::actor::code::init0};
-  map.set(Address::makeFromId(1),
-          {vm::actor::makeRawIdentityCid(std::string{code1}), head1})
-      .value();
+  // TODO(turuslan): remove after PR#415
```
Comment (Alexey-N-Chernyshov) [inline test/core/vm/state/state_tree_test.cpp:90]: 👁️

File: core/vm/actor/builtin/states/init_actor_state_raw.hpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    Hamt address_map{nullptr, 0, false};
+    uint64_t next_id{};
+    std::string network_name;
+
+    void load(const IpldPtr &ipld, const CID &_code) {
+      // TODO(turuslan): asActorCode from PR#415
```
Comment (Alexey-N-Chernyshov) [inline core/vm/actor/builtin/states/init_actor_state_raw.hpp:23]: 👁️

File: test/testutil/init_actor.hpp [hunk 4]
```diff
@@ -21,13 +21,22 @@ std::shared_ptr<fc::vm::state::StateTree> setupInitActor(
...
   auto store = state_tree->getStore();
-  fc::vm::actor::builtin::v0::init::InitActorState init_state;
-  init_state.address_map_0 = {store};
+  using namespace fc;
+  vm::actor::InitActorStateRaw init_state;
+  init_state.address_map = {store, 0, false};
   init_state.next_id = next_id;
   init_state.network_name = "n";
   EXPECT_OUTCOME_TRUE(head, store->setCbor(init_state));
+  // TODO(turuslan): remove after PR#415
```
Comment (Alexey-N-Chernyshov) [inline test/testutil/init_actor.hpp:30]: 👁️

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: test/testutil/init_actor.hpp [hunk 5]
```diff
@@ -21,13 +21,22 @@ std::shared_ptr<fc::vm::state::StateTree> setupInitActor(
         std::make_shared<fc::storage::ipfs::InMemoryDatastore>());
   }
   auto store = state_tree->getStore();
-  fc::vm::actor::builtin::v0::init::InitActorState init_state;
-  init_state.address_map_0 = {store};
+  using namespace fc;
```
Comment (wer1st) [inline test/testutil/init_actor.hpp:24]: Using `using namespace` in .h or .hpp is not good.
Please put this code in the necessary namespace

File: core/vm/actor/builtin/states/init_actor_state_raw.hpp [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+#pragma once
+
+#include "adt/address_key.hpp"
+#include "storage/hamt/hamt.hpp"
+#include "vm/actor/codes.hpp"
+
+namespace fc::vm::actor {
+  struct InitActorStateRaw {
```
Comment (wer1st) [inline core/vm/actor/builtin/states/init_actor_state_raw.hpp:13]: What is a destination of InitActorStateRaw?
It is inconsistent with other states

File: test/testutil/init_actor.hpp (hunk 5 above)
Comment (turuslan) [reply test/testutil/init_actor.hpp:24]: `using namespace` is discouraged in `namespace` level, this line affects only this function

Comment (wer1st) [reply test/testutil/init_actor.hpp:24]: Ok

**Review by wer1st — APPROVED** [review]

**Pull request merged**